cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/spark/SparkBatchSourceContext.java

```diff
@@ -82,6 +84,12 @@ public void setInput(String datasetName, Dataset dataset) {
     }
   }
 
+  @Override
+  public void setInput(Input input) {
+    Input trackableInput = ExternalDatasets.makeTrackable(sparkContext.getAdmin(), input);
```

Comment:
The other methods are for backwards compatibility. Since the other methods do not have the name of the input provider we cannot make it trackable.